src/main/java/fredbot/FredBot.java

```diff
@@ -0,0 +1,301 @@
+    private static int count;
+
+    private static final String COMMAND_DEADLINE = "deadline";
+    private static final String COMMAND_DELETE = "delete";
+    private static final String COMMAND_EVENT = "event";
+    private static final String COMMAND_EXIT = "bye";
+    private static final String COMMAND_LIST = "list";
+    private static final String COMMAND_MARK = "mark";
+    private static final String COMMAND_TODO = "todo";
+    private static final String COMMAND_UNMARK = "unmark";
+
+    private static final String MESSAGE_ADD = "Are you sure you'll ever get to it? Fine, I've added this task:";
+    private static final String MESSAGE_DELETE = "Finally lifting my load? I've removed this task:";
+    private static final String MESSAGE_EMPTY_LIST = "Go and touch some grass... your list is empty.";
+    private static final String MESSAGE_EMPTY_DESCRIPTION = "I can't do that if you don't give me the description...";
+    private static final String MESSAGE_ERROR = "What did you do... Something went wrong.";
+    private static final String MESSAGE_UNKNOWN_COMMAND = "I have no idea what you just said.";
+    private static final String MESSAGE_GOODBYE = "Finally... goodbye.";
+    private static final String MESSAGE_LIST_HEADER = "Mr Busy over here has these tasks in his list:";
+    private static final String MESSAGE_MARK = "Ok and do you want a medal for that? I've marked this as done:";
+    private static final String MESSAGE_UNMARK = "Why am I not surprised... I've marked this task as not done yet:";
+    private static final String MESSAGE_WELCOME = "Yo I'm FredBot. What do you want with me?";
+
+    private static final String PREFIX_BY = "/by";
+    private static final String PREFIX_FROM = "/from";
+    private static final String PREFIX_TO = "/to";
+
+    private static final String FILE_PATH = "data/fredbot.txt";
+
+    private static final Scanner SCANNER = new Scanner(System.in);
+
+    public static void main(String[] args) {
+        loadFredBot();
+        showWelcomeMessage();
+        while (true) {
+            String userInput = readUserInput();
+            executeCommand(userInput);
+            saveFredBot();
+        }
+    }
+
+    private static void loadFredBot() {
+        initTaskList();
+        File f = new File(FILE_PATH);
+        try {
+            loadTaskList(f);
+        } catch (FileNotFoundException e) {
+            initSaveFile(f);
+        }
+    }
+
+    private static void loadTaskList(File f) throws FileNotFoundException {
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+            String task = s.nextLine();
+            try {
+                processLine(task);
+            } catch (EmptyDescriptionException e) {
+                System.out.println(MESSAGE_EMPTY_DESCRIPTION);
+            }
+        }
+    }
+
+    private static void processLine(String task) throws EmptyDescriptionException {
+        String taskType = task.substring(0, 1);
```

Comment:
In this file, there are quite a fair number of [Magic Literals](https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers).

Avoid this, as these will make debugging a headache moving on. You can refactor them as constants. For example, instead of having the literal `"D"`, you can refactor as `"TASK_DEADLINE"`. This makes the code much more readable.

A good heuristic is to think in terms of a 'third party' that knows nothing about your codebase. Would they understand at a glance what `2` in `taskArgs[2]` is, or what `"T"` stands for?